variable/variable.cpp

```diff
@@ -123,7 +123,36 @@ Variable Variable::slice(const Slice params) const {
   return out;
 }
 
+void Variable::validateSlice(const Slice &s, const Variable &data,
+                             const std::string &parent_key) const {
```

Comment:
This is a really odd member, I don't think it should exist with this signature. Can we keep the checks that require the parent key in `Dict::validateSlice`?

Reply:
Agreed. This does not make good sense. Will change.